### ISSUE TITLE
Unparses `foo` as `#{"foo"}`

### ISSUE THREAD

**Issue by eiko** [mbj/unparser#82]
Unparser unparses \`foo\` as \`#{"foo"}\` which forms a different parse tree than the original source:

\`foo\` == xstr { str { 'foo' } }
\`#{"foo"}\` == xstr { begin { str { 'foo' } } }

While the two code snippets are functionally the same in standard ruby, Opal interprets them differently, so the distinction matters. Is there any reason Unparser can't differentiate between the two to create the proper tree?

**Comment by mbj** [comment]
> While the two code snippets are functionally the same in standard ruby,

This is (so far) a standard ruby unparser. I never intend to explicitly support opal.

>  Is there any reason Unparser can't differentiate between the two to create the proper tree?

Technically no. Practically yes. Elaboration: I do this library on my free time and that one is super limited.

All my OSS is driven by clients needs, if a clients project stumbles about a problem in my libs they get a patch, if not: nothing happens.

I'd accept patches for this one.

**Comment by hovissimo** [comment]
Is this really intended behavior?

```ruby
irb(main):038:0> initial_ast = Unparser.parse('"My name is #{first_name} #{last_name}."')    
=> s(:dstr,                                                                                  
  s(:str, "My name is "),                                                                    
  s(:begin,                                                                                  
    s(:send, nil, :first_name)),                                                             
  s(:str, " "),                                                                              
  s(:begin,                                                                                  
    s(:send, nil, :last_name)),                                                              
  s(:str, "."))                                                                              

irb(main):039:0> round_trip_ast = Unparser.parse(Unparser.unparse(initial_ast))
=> s(:dstr,                                                                    
  s(:begin,                                                                    
    s(:str, "My name is ")),                                                   
  s(:begin,                                                                    
    s(:send, nil, :first_name)),                                               
  s(:begin,                                                                    
    s(:str, " ")),                                                             
  s(:begin,                                                                    
    s(:send, nil, :last_name)),                                                
  s(:begin,                                                                    
    s(:str, ".")))                                                             

irb(main):040:0> initial_ast == round_trip_ast                                 
=> false                                                                       
```

**Comment by mbj** [comment]
@hovissimo Not intended.

The #129 branch is attempting to clean it up as a side effect.

**Comment by mbj** [comment]
Sorry did not intend to close, was fat fingering.

**Comment by mbj** [comment]
Very likely this gets fixed with #148
